f3/granite.go

```diff
@@ -397,7 +509,49 @@ func (i *instance) beginConverge() {
 	i.phase = CONVERGE_PHASE
 	ticket := i.vrf.MakeTicket(i.beacon, i.instanceID, i.round, i.participantID)
 	i.phaseTimeout = i.alarmAfterSynchrony(CONVERGE_PHASE.String())
-	i.broadcast(i.round, CONVERGE_PHASE, i.proposal, ticket)
+	prevRoundState := i.roundState(i.round - 1)
+	var justification Justification
+	var ok bool
+	if prevRoundState.committed.HasStrongQuorumAgreement(ZeroTipSetID()) {
+		value := ECChain{}
+		signers := prevRoundState.committed.getSigners(value)
+
+		signatures := prevRoundState.committed.getSignatures(value, signers)
+		aggSignature := make([]byte, 0)
+		for _, sig := range signatures {
+			aggSignature = i.host.Aggregate([][]byte{sig}, aggSignature)
+		}
+		justification = Justification{
+			Instance:  i.instanceID,
+			Round:     i.round - 1,
+			Step:      COMMIT_PHASE.String(),
+			Value:     value,
+			Signers:   signers,
+			Signature: aggSignature,
+		}
```

Comment:
This should disappear if we adopt the GMessage changes I suggested.

Reply:
I am not quite sure how this would "disappear"? The justification still needs to be created and the signatures aggregated.

Reply:
by that, I meant that the SignedMessage could have been stored in the prevRoundState but it is not that important